### ISSUE TITLE
Unit 5 and virtual pet starter -- language inconsistency on what students will do

### ISSUE THREAD

**Issue by kfisler** [bootstrapworld/curr#200]
The virtual pet starter file has initial versions of the next-state-tick and next-state-key functions, both of which already have if statements in them.

The unit text, in contrast, says that students will write these functions (hinting that they will do so from scratch).

I would have expected the starter file to at most just return the given a-pet as the output of the two next-state functions.  Why is the if-scaffolding in there?

More generally, we need to decide how much students are doing from scratch so we can align the starter file and the unit text.

(I'm also personally confused why the code doesn't result in anything happening -- shouldn't the initial next-state-tick be producing new states, which would cause the bars to get shorter until the pet dies?)

**Comment by Emmay** [comment]
Didn't notice the inconsistency there, I'm happy to change up the starter file- we'll also want a specific section of the lesson to talk about the has-lost function if it's not used in the starter. However, the code as written does make the bars get shorter in the animation, which starts automatically.

**Comment by kfisler** [comment]
Sorry -- I missed the bars the first time I tried viewing this -- my bad.

Joe: are you taking care of the unit 5 issues that we are flagging, or should one of us do it?

**Comment by jpolitz** [comment]
Assigned myself.  I put in the checks in the starter code because otherwise the draw function tries to show a negative width for the bars, ending in error, which isn't very instructive.

Another way to handle this is to use `num-max` with `0` in the draw function to simply dodge it without a branch in the program itself.

Then it's much easier to just write the tick and key functions from scratch.

**Comment by jpolitz** [comment]
@Emmay – are you currently owning the starter file or managing it somehow?  Or do you just make new versions?  Don't want to clobber anything.

We could also share ownership (via docs) of one file that's the public version and make edits to that.

**Comment by Emmay** [comment]
Yes, I have the starter file, which I can share with you via docs so you can make edits too. I assume if I share the doc in my code.pyret.org/shared folder (as opposed to the code.pyret.org folder), that will allow you to edit the file that's linked directely in the lessons, correct?

**Comment by jpolitz** [comment]
Yup!  Just share https://docs.google.com/file/d/0B9rKDmABYlJVXy00M1VteEZxaHM/edit with me (or replace with a more recent share URL).

We can also work from 

code.pyret.org/editor?program=0B9rKDmABYlJVXy00M1VteEZxaHM

And saving to that will update the shared version, so we don't have to make copies.

**Comment by jpolitz** [comment]
OK, I updated the starter file.  I _removed_ the subtractions in `next-state-tick`, which I intended to have students write.  Re: https://github.com/bootstrapworld/curr/issues/190, which is related, I think we should give them a worked version of the animation design sheet that they can follow and copy into the code.

Then they do the next-state-tick with the sheet partially or minimally started.

Then they write `has-lost` on their own and use it – it's not so bad if they write all 3 functions as a conditional that copy-pastes the checks for `> 0` on all three fields, or they could write it with a helper function.  Since some teachers won't, we get an extra moment to talk about helper functions if we want it.
